pandas/core/apply.py

```diff
@@ -471,8 +471,22 @@ def compute_dict_like(
 
                 keys += [key] * len(key_data)
                 results += key_data
-        else:
+        elif is_groupby:
             # key used for column selection and output
+
+            df = obj.obj
```

Comment:
Use `selected_obj`, `obj.obj` contains all the columns of the input.

```
df = pd.DataFrame({'a': [1], 'b': 2, 'c': 3})
gb = df.groupby('a')[['b']]
print(gb.obj.columns)
# Index(['a', 'b', 'c'], dtype='object')  <-- wrong
print(gb._selected_obj.columns)
# Index(['b'], dtype='object')  <-- right

```

Reply:
resolved, the original code (now in the else branch) uses `(obj._gotitem(key, ndim=1)` which internally uses `self.obj`, equivalent to `obj.obj`. is that ok since any column that does not exists is already check for beforehand?

Reply:
Indeed, disabling the check you mentioned results in this behavior:

```
df = pd.DataFrame({"a": [1, 1, 2], "b": 3, "c": 4})
gb = df.groupby("a")[["b"]]
print(gb.agg({"b": "sum", "c": "mean"}))
#    b    c
# a        
# 1  6  4.0
# 2  3  4.0
```